tests/storages_tests/test_storages.py

```diff
@@ -166,65 +169,38 @@ def test_set_and_get_study_directions(storage_mode: str) -> None:
 
     with StorageSupplier(storage_mode) as storage:
 
-        for target, opposite in [
-            ((StudyDirection.MINIMIZE,), (StudyDirection.MAXIMIZE,)),
-            ((StudyDirection.MAXIMIZE,), (StudyDirection.MINIMIZE,)),
-            (
-                (StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE),
-                (StudyDirection.MAXIMIZE, StudyDirection.MINIMIZE),
-            ),
-            (
-                [StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE],
-                [StudyDirection.MAXIMIZE, StudyDirection.MINIMIZE],
-            ),
+        for target in [
+            (StudyDirection.MINIMIZE,),
+            (StudyDirection.MAXIMIZE,),
+            (StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE),
+            (StudyDirection.MAXIMIZE, StudyDirection.MINIMIZE),
+            [StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE],
+            [StudyDirection.MAXIMIZE, StudyDirection.MINIMIZE],
         ]:
-            study_id = storage.create_new_study()
+            study_id = storage.create_new_study(directions=target)
 
-            def check_set_and_get(directions: Sequence[StudyDirection]) -> None:
-                storage.set_study_directions(study_id, directions)
+            def check_get() -> None:
```

Comment:
[nit] This is perhaps an out-of-scope comment, but I think we don't need to define an inner function here.
`check_get()` function cannot reduce the number of lines and cannot make this test readable.

We can work it in the following pull request.